src/components/menu/menu.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { FC, MutableRefObject, PropsWithChildren } from 'react'
+import { useClickOutside } from '../../hooks/use-click-outside'
+import { MenuItemGroup } from './menu.atoms'
+import { MenuProvider, useMenu } from './provider'
+import { ElMenu, ElMenuItemContainer, ElMenuPopover } from './styles'
+
+const Trigger = ({ children }) => {
+  const { triggerProps, isOpen } = useMenu()
+
+  return children({ triggerProps, isOpen })
+}
+
```

Comment:
**suggestion:** I think this `Popover` is doing enough to warrant it's own file (much like we have a separate file for the context/provider.